### PR TITLE
Light and dark mode with toggle

### PR DESCRIPTION
**Pull request by AbhijeetSingla**
# About
Added dark and light mode with a toggle and local storage in javascript to save user preference.

# Preview
[live-link](https://as1553-component-library-experimental.netlify.app)

### CODE REVIEW

File: css/color-scheme.css [hunk 1]
```diff
@@ -95,6 +23,21 @@ body {
 .flex { display: flex; gap: var(--gap, 1rem);}
 .grid { display: grid; gap: var(--gap, 1rem);}
 
+/* theme toggle button */
+
+#theme-toggle-button {
+    position: absolute;
+    top: 0;
+    right: 0;
+    height: 40px;
```
Comment (dev-ashishsethi) [inline css/color-scheme.css:32]: for sizes greater than 16px use rem

File: css/color-scheme.css [hunk 2]
```diff
@@ -455,6 +429,22 @@ body {
...
+
+/* code */
+
+.code-base {
+    --max-spc-width: 600px;
+    margin-inline-start: 0;
+}
+
+.code-base>figcaption {
+    font-size: 0.55rem;
```
Comment (dev-ashishsethi) [inline css/color-scheme.css:441]: for sizes less than 1rem use pixels

File: css/color-scheme.css [hunk 3]
```diff
@@ -455,6 +429,22 @@ body {
...
+    padding-block-start: 1rem;
+    padding-inline-start: 1.2rem;
+    padding-inline-end: 1.2rem;
```
Comment (dev-ashishsethi) [inline css/color-scheme.css:444]: could have used padding-inline

File: css/color-scheme.css [hunk 4]
```diff
@@ -379,6 +336,22 @@ body {
...
+.typography-base {
+    text-align: center;
+    flex-direction: column;
+    --gap: 0;
+}
+
+/* aside */
+
+.aside-base {
+    color: var(--red);
```
Comment (dev-ashishsethi) [inline css/color-scheme.css:350]: give fallback values as well

File: css/color-scheme.css [hunk 5]
```diff
@@ -95,6 +23,21 @@ body {
...
 
+/* theme toggle button */
+
+#theme-toggle-button {
+    position: absolute;
+    top: 0;
+    right: 0;
+    height: 40px;
+    width: 40px;
+    background-color: transparent;
```
Comment (dev-ashishsethi) [inline css/color-scheme.css:34]: could have used background instead of writing background-color and background -repeat separately

**Review by dev-ashishsethi — APPROVED** [review]
do check comments

File: css/color-scheme.css (hunk 4 above)
Comment (AbhijeetSingla) [reply css/color-scheme.css:350]: noted

File: css/color-scheme.css (hunk 2 above)
Comment (AbhijeetSingla) [reply css/color-scheme.css:441]: noted

File: css/color-scheme.css (hunk 1 above)
Comment (AbhijeetSingla) [reply css/color-scheme.css:32]: noted

File: css/color-scheme.css (hunk 3 above)
Comment (AbhijeetSingla) [reply css/color-scheme.css:444]: Some browsers do not support short hands or sometimes bug out, so I preferred this way

**Pull request closed without merging**